cirq-core/cirq/circuits/circuit.py

```diff
@@ -2142,34 +1996,25 @@ def insert_into_range(self, operations: 'cirq.OP_TREE', start: int, end: int) ->
             raise IndexError(f'Bad insert indices: [{start}, {end})')
 
         flat_ops = list(ops.flatten_to_ops(operations))
-        for op in flat_ops:
-            self._device.validate_operation(op)
 
         i = start
         op_index = 0
         cannot_add_lambda = lambda a, b: b.operates_on(a.qubits)
-        if self._device != cirq.UNCONSTRAINED_DEVICE:
-            _compat._warn_or_error(
-                "In Cirq v0.15 device specific validation in "
-                "insert_into_range will no longer enforced."
-            )
-            cannot_add_lambda = lambda a, b: not self._device.can_add_operation_into_moment(a, b)
 
-        with _compat.block_overlapping_deprecation('(can_add_operation_into_moment|insert)'):
-            while op_index < len(flat_ops):
-                op = flat_ops[op_index]
-                while i < end and cannot_add_lambda(op, self._moments[i]):
-                    i += 1
-                if i >= end:
-                    break
+        while op_index < len(flat_ops):
+            op = flat_ops[op_index]
+            while i < end and cannot_add_lambda(op, self._moments[i]):
```

Comment:
nit: Since now this is the only usage of `cannot_add_lambda`, replace with `not self._moments[i].operates_on(op.qubits)` for clarity.

Reply:
Done.